support-workers/src/main/scala/com/gu/emailservices/ContributionEmailFields.scala

```diff
@@ -1,52 +1,62 @@
 package com.gu.emailservices
 
 import com.gu.emailservices.SubscriptionEmailFieldHelpers.{formatDate, hyphenate, mask}
+import com.gu.salesforce.Salesforce.SfContactId
 import com.gu.support.workers._
+import com.gu.support.workers.states.ProductTypeCreated.ContributionCreated
 import org.joda.time.DateTime
 
-object ContributionEmailFields {
+import scala.concurrent.{ExecutionContext, Future}
 
-  def build(
-    allProductsEmailFields: AllProductsEmailFields,
-    created: DateTime,
-    amount: BigDecimal,
-    paymentMethod: PaymentMethod
-  ): EmailFields = {
-    import allProductsEmailFields._
+class ContributionEmailFields(
+  getMandate: String => Future[Option[String]],
```

Comment:
before, the getMandate was done at the top level.  But now we have different products, some of which don't have payment methods.
The low intervention solution would be to still run it at the top level by making the pattern match more comprehensive.
Since by the time we actually need the value, we already know whether it's DD, I decided to push it down to the point of need.  Then we only concern ourselves about it in one place, however we do need to pass in the mandate service, and we do need to return a `Future` (or at least `F` where `F` is a `Functor`)

There is undoubtedly an opprtunity to make getMandate into a proper Service, and probably reduce the duplication a bit between cont/DS/paper+weekly.  I will see whether to do it in this PR or save for later.